### PR TITLE
Fix vaadin dialog full size

### LINKED ISSUE #103: setSizeFull doesn't work
Use case: full-screen dialogs (mobile).

Can't add class names to the dialog, as it doesn't implement `HasStyle`.

Via `dialog.getElement().getClassList().add("custom");` you can add class names to the dialog, but not the overlay (nor the component renderer). Thus, I have to use `position: fixed;` on the actual contents of the dialog to make it fullscreen.

### LINKED ISSUE #142: Add support for non-modal dialogs
Please add support for non-modal dialogs. My use-case is that an user is editing a web page, and she should be able to see the preview of the web page in a non-modal dialog (so that she can interact with the editor components).

### PR DESCRIPTION
**Pull request by shadikhani**
Fix https://github.com/vaadin/vaadin-dialog/issues/142
Fix #103

### CODE REVIEW

File: vaadin-dialog-flow-integration-tests/src/test/java/com/vaadin/flow/component/dialog/tests/DialogTestPageIT.java [hunk 1]
```diff
@@ -205,6 +204,26 @@ public void openEmptyDialog_dialogContentHasWidth() {
...
+                By.id("overlay"));
+        Assert.assertTrue(
+                overlay.getAttribute("style").contains("width: 100%;"));
+        Assert.assertTrue(
+                overlay.getAttribute("style").contains("height: 100%;"));
+
+        WebElement content = overlay.findElement(By.id("content"));
+
+        Assert.assertTrue(
+                content.getAttribute("style").contains("width: 100%;"));
```
Comment (tomivirkki) [inline vaadin-dialog-flow-integration-tests/src/test/java/com/vaadin/flow/component/dialog/tests/DialogTestPageIT.java]: The fact that the element has style `height: 100%` doesn't prove it's actually taking full size. You'll need to get the effective size of the element.

Comment (shadikhani) [reply vaadin-dialog-flow-integration-tests/src/test/java/com/vaadin/flow/component/dialog/tests/DialogTestPageIT.java]: Done.

File: vaadin-dialog-flow-integration-tests/src/test/java/com/vaadin/flow/component/dialog/tests/DialogTestPageIT.java [hunk 2]
```diff
@@ -205,6 +205,34 @@ public void openEmptyDialog_dialogContentHasWidth() {
...
+
+        String paddingWidth = content.getCssValue("padding");
+        int paddingValue = Integer
+                .valueOf(paddingWidth.substring(0, paddingWidth.length() - 2));
+
+        String divWidth = div.getCssValue(ElementConstants.STYLE_WIDTH);
+        int divValue = Integer
+                .valueOf(divWidth.substring(0, divWidth.length() - 2));
+
+        Assert.assertEquals(overlayValue - paddingValue * 2, divValue);
```
Comment (tomivirkki) [inline vaadin-dialog-flow-integration-tests/src/test/java/com/vaadin/flow/component/dialog/tests/DialogTestPageIT.java]: We need to run the same tests for height.

Note: I wouldn't guess from this line what "overlayValue" means. Rename to "overlayWidthValue" or something

Comment (shadikhani) [reply vaadin-dialog-flow-integration-tests/src/test/java/com/vaadin/flow/component/dialog/tests/DialogTestPageIT.java]: Done.

File: vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java [hunk 3]
```diff
@@ -433,6 +438,10 @@ private void attachComponentRenderer() {
                 "<flow-component-renderer appid=\"%s\" nodeid=\"%s\"></flow-component-renderer>",
                 appId, nodeId);
         template.setProperty("innerHTML", renderer);
-    }
 
+        getElement().executeJs("this.$.overlay.$.overlay.style.height=$0",
```
Comment (tomivirkki) [inline vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java]: Discussed this in person.`setWidth/Height` would produce "24px" (Lumo padding) smaller outcome with this approach alone.

In order not to break the existing functionality, we'll also need to add the effective padding size to the value applied to the element. But this is only the case with non-relative units; `setWidth(100%)` should result in full width, not full-width + padding.

Comment (tomivirkki) [reply vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java]: New decision related to this, we'll instead go with the current revision of this PR and consider the new functionality as fix for https://github.com/vaadin/vaadin-dialog-flow/issues/144

**Pull request merged**